static_frame/core/index_hierarchy.py

```diff
@@ -943,24 +946,14 @@ def isin(self, other: tp.Iterable[tp.Iterable[tp.Hashable]]) -> np.ndarray:
         for seq in other:
             if not hasattr(seq, '__iter__'):
                 raise RuntimeError('must provide one or more iterables within an iterable')
-            # must use iterable to array to properly handle heterogenous types, or if already an array
-            v, _ = iterable_to_array(seq)
-            # if seq is a tuple, could check self.keys()
-            if len(v) == self.depth:
-                matches.append(v)
+            if len(seq) == self.depth:
```

Comment:
I think we still need lines 946-949, since `seq` could be an unsized iterator.

Reply:
That is a good point. I will fix that.